fabtests/functional/inj_complete.c

```diff
@@ -149,37 +160,41 @@ int main(int argc, char **argv)
 
 	opts = INIT_OPTS;
 	use_sendmsg = 0;
-	send_inj_count = 1;
+	opts.iterations = 1;
 
 	hints = fi_allocinfo();
 	if (!hints)
 		return EXIT_FAILURE;
 
-	while ((op = getopt(argc, argv, "m:OSvh" ADDR_OPTS INFO_OPTS)) != -1) {
+	while ((op = getopt(argc, argv, "NvhA:" CS_OPTS ADDR_OPTS INFO_OPTS)) != -1) {
 		switch (op) {
 		default:
 			ft_parse_addr_opts(op, optarg, &opts);
 			ft_parseinfo(op, optarg, hints, &opts);
+			ft_parsecsopts(op, optarg, &opts);
 			break;
-		case 'O':
-			send_inj_count = 100;
-			break;
-		case 'm':
-			opts.transfer_size = strtoul(optarg, NULL, 0);
-			break;
-		case 'S':
+		case 'N':
 			use_sendmsg = 1;
 			break;
 		case 'v':
 			opts.options |= FT_OPT_VERIFY_DATA;
 			break;
+		case 'A':
+			if (!strncasecmp("inj_complete", optarg, 12)) {
+				flag = FI_INJECT_COMPLETE;
+			} else if (!strncasecmp("inject", optarg, 6)) {
+				flag = FI_INJECT;
+			} else {
+				printf ("Unsupported flag\n");
+				return EXIT_FAILURE;
+			}
+			break;
 		case '?':
 		case 'h':
-			ft_usage(argv[0], "FI_Inject_Completion Functional Test");
-			FT_PRINT_OPTS_USAGE("-m <size>", "size of Injection message");
-			FT_PRINT_OPTS_USAGE("-S", "enable testing with fi_sendmsg");
-			FT_PRINT_OPTS_USAGE("-O", "enable testing injection overrun");
+			ft_csusage(argv[0], "FI_Inject_Completion Functional Test");
+			FT_PRINT_OPTS_USAGE("-N", "enable testing with fi_sendmsg");
```

Comment:
This patch should also change the -S in runfabtests to be -N since you change the argument in this patch